packages/twenty-front/src/pages/settings/workspace/SettingsCustomDomainEffect.tsx

```diff
@@ -1,48 +1,14 @@
-import { currentWorkspaceState } from '@/auth/states/currentWorkspaceState';
-import { useEffect, useCallback } from 'react';
-import { useSetRecoilState, useRecoilValue } from 'recoil';
-import { isDefined } from 'twenty-shared';
-import { useCheckCustomDomainValidRecordsMutation } from '~/generated/graphql';
-import { customDomainRecordsState } from '~/pages/settings/workspace/states/customDomainRecordsState';
+import { useEffect } from 'react';
+import { useCheckCustomDomainValidRecords } from '~/pages/settings/workspace/hooks/useCheckCustomDomainValidRecords';
 
 export const SettingsCustomDomainEffect = () => {
-  const [checkCustomDomainValidRecords] =
-    useCheckCustomDomainValidRecordsMutation();
-
-  const setCustomDomainRecords = useSetRecoilState(customDomainRecordsState);
-
-  const currentWorkspace = useRecoilValue(currentWorkspaceState);
-
-  const checkCustomDomainValidRecordsPolling = useCallback(async () => {
-    setCustomDomainRecords((currentState) => ({
-      ...currentState,
-      loading: true,
-    }));
-    checkCustomDomainValidRecords({
-      onCompleted: (data) => {
-        if (isDefined(data.checkCustomDomainValidRecords)) {
-          setCustomDomainRecords({
-            loading: false,
-            customDomainRecords: data.checkCustomDomainValidRecords,
-          });
-        }
-      },
-    });
-  }, [checkCustomDomainValidRecords, setCustomDomainRecords]);
+  const { checkCustomDomainRecords } = useCheckCustomDomainValidRecords();
 
   useEffect(() => {
-    let pollIntervalFn: null | ReturnType<typeof setInterval> = null;
-    if (isDefined(currentWorkspace?.customDomain)) {
-      checkCustomDomainValidRecordsPolling();
-      pollIntervalFn = setInterval(checkCustomDomainValidRecordsPolling, 6000);
-    }
-
-    return () => {
-      if (isDefined(pollIntervalFn)) {
-        clearInterval(pollIntervalFn);
-      }
-    };
-  }, [checkCustomDomainValidRecordsPolling, currentWorkspace?.customDomain]);
+    checkCustomDomainRecords();
```

Comment:
If it only has to be run once, let's use a state instead of relying on a useEffect which is unpredictable on the long run and opens the door to future nasty side effects.

A simple atom like isCustomDomainChecked is ok